ras/vpa_MaxMem_check.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2021 IBM
+# Author: Shirisha Ganta <shirisha.ganta1@ibm.com>
+
+from avocado import Test
+from avocado.utils import process, genio, distro
+from avocado import skipIf
+
+IS_POWER_NV = 'PowerNV' in genio.read_file('/proc/cpuinfo').rstrip('\t\r\n\0')
+
+
+class Cpu_VpaData(Test):
+
+    '''
+    checks for cpu files and MaxMem
+    '''
+
+    @skipIf(IS_POWER_NV, "This test is supported on PowerVM environment")
+    def setUp(self):
+        detected_distro = distro.detect()
+        if detected_distro.name not in ['rhel', 'SuSE']:
+            self.cancel("Test case is supported only on RHEL and SLES")
+        
+    def test(self):
+
+        self.log.info("===Checking for cpu VPA data and MaxMem==")
+        process.system('ls /sys/kernel/debug/powerpc/vpa')
+        cpu_count = process.system_output("ls /sys/kernel/debug/powerpc/vpa | wc -l", shell=True)
+        process.system('cat /proc/powerpc/lparcfg')
+        maxmem = process.system_output("cat /proc/powerpc/lparcfg | "
+                                       "grep MaxMem | "
+                                       "cut -d'=' -f2", shell=True).strip()
+        total_cpus = process.system_output("lscpu | awk '{if(NR==3) print $2}'", shell=True)
```

Comment:
please use the python `re` module to do the operation as embedding  bash command in python ,not a good practice as it use subprocess module it kind of more over head for CPU